### PR TITLE
[DOC] Restructure Minikube Installation

### PR DESCRIPTION
**Pull request by alero-awani**
This is still a work in progress. 

A lot of changes need to be made in this Documentation, or it might confuse users. Specifically at what points the `mesheryctl system config minikube` command should be run.

**Notes for Reviewers**

**Checklist**
- [x] Pre-flight checks.
- [x] Installing Meshery in-cluster. [PR](https://github.com/meshery/meshkit/pull/588/files)
- [x] Installing Meshery out-of-cluster (using docker).
- [x] Installing Meshery out-of-cluster and connecting it to a Minikube cluster that uses a docker-driver.
       - Users will need to modify their kubeconfig and add in the Minikube external API server address.
- [x] Installing Meshery out-of-cluster and connecting it to a Minikube cluster that uses a different driver.
      - Tested with hyperkit
- [x] Uploading the Kubeconfig file directly in the UI.
      -They can upload directly without running minify and flatten command. This [PR](https://github.com/meshery/meshery/pull/11987) fixes that.
- [x] Post-Installation steps
- [x] Users should confirm in the UI that their Configuration files have been uploaded and do a ping test.
- [x] Point them to a doc on how they can start deployments or use Visualizer.

Other 

- [x] Proper Introduction for all sections.

**[Signed commits](https://github.com/meshery/meshery/blob/master/CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [x] Yes, I signed my commits.

### CODE REVIEW

**Comment by l5io** [comment]
🚀 Preview for commit 4f9834b880b300145bf9c60c72bff589e62c558a at: https://66e318740c6395428ec7c123--meshery-docs-preview.netlify.app

File: docs/pages/installation/minikube.md
Comment (jamieplu) [inline docs/pages/installation/minikube.md:78]: Oh, interesting. The `system config minikube` command requires an active token, huh?

Comment (Alero-Awani) [reply docs/pages/installation/minikube.md:78]: Hi @jamieplu, yes it does

**Comment by l5io** [comment]
🚀 Preview for commit f64ffa20e7d9a1746a78e0e6ff8d1072d523bf42 at: https://66e8121877c5c91b88747e95--meshery-docs-preview.netlify.app

File: docs/pages/installation/minikube.md
Comment (leecalcote) [reply docs/pages/installation/minikube.md:78]: @Alero-Awani why?

Comment (Alero-Awani) [reply docs/pages/installation/minikube.md:78]: My apologies, The command requires authentication from either an active token or an existing logged-in session. The token is optional.

Comment (leecalcote) [reply docs/pages/installation/minikube.md:78]: Thanks, no sweat. Why do we require users to have an active session with Meshery Server in order to execute this command?

**Review by leecalcote — COMMENTED** [review]
in the updated Minikube installation guide, you can add reference to https://docs.meshery.io/guides/troubleshooting/installation#setting-up-meshery-using-kind-or-minikube, if it isn’t present already.

File: docs/_includes/installation/accessing-meshery-ui.md
Comment (leecalcote) [inline docs/_includes/installation/accessing-meshery-ui.md:37]: Good description.

File: docs/_includes/installation/accessing-meshery-ui.md
Comment (leecalcote) [inline docs/_includes/installation/accessing-meshery-ui.md:41]: An additional note regarding pre-selecting a Provider in order to control which identity providers are authorized would be good to include in a Production Deployment doc.

**Review by leecalcote — APPROVED** [review]

File: docs/_includes/installation/accessing-meshery-ui.md
Comment (leecalcote) [reply docs/_includes/installation/accessing-meshery-ui.md:41]: Reference - https://github.com/meshery/meshery/issues/2451

**Comment by l5io** [comment]
🚀 Preview for commit d987ee67f06c0fd4d6ebc06635925e0b58d6f9e9 at: https://66ea04062226f5a7748ada03--meshery-docs-preview.netlify.app

**Comment by l5io** [comment]
🚀 Preview for commit ef179bde90de9f17ea6038965bb4084b4a3f32e2 at: https://671f481874664d7f28b722a7--meshery-docs-preview.netlify.app

**Comment by Alero-Awani** [comment]
@leecalcote this has the ordered steps and can be merged once approved.

**Comment by l5io** [comment]
🚀 Preview for commit 3479721c853ace7a7e8d6d6ff3ee803221e64918 at: https://671f5618247306a4a686fbe0--meshery-docs-preview.netlify.app

File: docs/pages/installation/minikube.md
Comment (Alero-Awani) [reply docs/pages/installation/minikube.md:78]: @leecalcote, I tried to confirm with Uzair before he left. But from what I have seen. This command makes a request to the Meshery Server and retrieves the existing contexts and then uploads the current context to Meshery. So it needs to authenticate with the Meshery Server first. In a situation where the user doesn't run `mesheryctl system login` they get a response that shows they need to login first.

**Review by Aijeyomah — APPROVED** [review]

File: docs/pages/installation/minikube.md
Comment (leecalcote) [reply docs/pages/installation/minikube.md:78]: @Alero-Awani a chicken-before-the-egg problem. // @lekaf974

**Comment by Alero-Awani** [comment]
@lekaf974 Here's the updated documentation for the Minikube install page.

**Comment by Alero-Awani** [comment]
@leecalcote While the code structure for `mesheryctl system config` is still being discussed. I believe this should be merged to quickly help newcomers trying to setup in the meantime.

**Pull request merged**
